app/models/user.rb

```diff
@@ -22,12 +28,38 @@ class User < RedisStore
   # Add additional MVI attributes
   alias redis_key uuid
 
+  # mvi 'golden record' data
+  attribute :mvi
+
   validates :uuid, presence: true
   validates :email, presence: true
 
+  after_initialize :fetch_mvi_data
+
   def self.sample_claimant
     attrs = JSON.load(ENV['EVSS_SAMPLE_CLAIMANT_USER'])
     attrs[:last_signed_in] = Time.now.utc
     User.new attrs
   end
+
+  def fetch_mvi_data
+    given_names = [first_name]
+    given_names.push middle_name unless middle_name.nil?
+    message = MVI::Messages::FindCandidateMessage.new(
+      given_names,
+      last_name,
+      dob,
+      ssn,
+      gender
+    )
+    if message.valid?
+      response = MVI_SERVICE.find_candidate(message)
+      update(mvi: response)
+    else
+      errors = message.errors.full_messages.join(', ')
+      Rails.logger.warn "MVI user data not retrieved: invalid message: #{errors}"
+    end
+  rescue MVI::ServiceError => e
```

Comment:
Wasn't sure if not having MVI data right after a user was initialized needed to be raised, the user is valid and authorized at that point but not later on for things that require MVI ids. Might be a sign that we need to fetch MVI data at that point rather than on user initialization.